src/babashka/curl.clj

```diff
@@ -133,27 +134,31 @@
 ;;;; End utils
 
 ;;;; Response Parsing
-(defn- assoc-if-not-empty [map key val]
-  (if (empty? val)
-    map
-    (assoc map key val)))
-
-(defn- input-stream-eof? [^java.io.InputStream rdr]
-  (.mark rdr 1)
-  (let [eof? (= -1 (.read rdr))]
-    (when-not eof?
-      (.reset rdr))
-    eof?))
+
+;; See https://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/java/io/DataInputStream.java#l501
+(defn- read-line [^PushbackInputStream is]
+  (let [sb (StringBuilder.)
+        res
+        (loop []
+          (let [c1 (.read is)]
+            (case c1
+              ;; 10 = \n, 13 = \r
+              (-1 10) (str sb)
+              13 (let [c2 (.read is)]
+                   (when (and (not= c2 10) (not= c2 -1))
+                     (.unread is c2))
+                   (str sb))
+              (do (.append sb (char c1))
+                  (recur)))))]
+    res))
```

Comment:
What do you think about adjusting the code slightly. I think this might be more clear that the function returning the contents of the string buffer. Thoughts? 

```suggestion
(defn- read-line [^PushbackInputStream is]
  (let [sb (StringBuilder.)]
    (loop []
      (let [c1 (.read is)]
        (case c1
          ;; 10 = \n, 13 = \r
          (-1 10) :done
          13 (let [c2 (.read is)]
               (when (and (not= c2 10) (not= c2 -1))
                 (.unread is c2)))
          (do (.append sb (char c1))
              (recur)))))
    (str sb)))
```

Reply:
I'm going to commit this suggestion and we can continue the discussion on the main PR, rather than have it split over multiple repositories.